robottelo/rerun_failures/uncollector.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+import pytest
+
+from robottelo.config import settings
+from robottelo.report_portal.portal import ReportPortal
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_launch(rp):
+    """Returns the latest launch of satellite"""
+    _version = settings.server.version
+    sat_version = f'{_version.base_version}.{_version.epoch}'
+    launch = next(iter(rp.launches(sat_version=sat_version).values()))
+    # Stop session based on RP Launch statistics and info
+    pass_percent = round(
+        100 * (int(launch.statistics['passed']) / int(launch.statistics['total']))
+    )
+    if launch.info['isProcessing']:
+        pytest.skip(f'The latest launch of satellite version {sat_version} is not Finished yet.')
```

Comment:
Reviving the discussion that started in #7758 I don't think this is a good pattern for the hook methods, as pytest.skip is relevant to a single test item, not the collection hook for the entire session.

Following your intended design of stopping the test session in either of these cases, I think that this method should raise an exception, and that exception should be logged/re-raised within the hook to stop the test session.

Reply:
I moved that back to the hook, but again the hooking is looking messy ! :)